src/library/features/crates/cratetreemodel.h

```diff
@@ -0,0 +1,43 @@
+#ifndef CRATE_TREE_MODEL_H
+#define CRATE_TREE_MODEL_H
+
+#include <QList>
+#include <QMap>
+
+#include "library/treeitemmodel.h"
+#include "library/features/crates/cratehierarchy.h"
+#include "library/crate/crate.h"
+
+// This class represents the crate tree for the nested crates feature
+
+class CrateTreeModel : public TreeItemModel {
+  public:
+    CrateTreeModel(LibraryFeature* pFeature,
+                   TrackCollection* pTrackCollection,
+                   CrateHierarchy* pCrateHierarchy);
+    //virtual ~CrateTreeModel();
+
+    // QVariant data(const QModelIndex& index, int role) const override;
+    // bool setData(const QModelIndex& index, const QVariant& value, int role) override;
+
+  public slots:
+    void reloadTree() override;
+
+  protected:
+
+  private:
+    bool findParentChildFromPath(Crate& parent,
```

Comment:
Wait, why is this a member of CrateTreeModel? Wouldn't it make more sense as a member of CrateStorage or CrateHierarchy?